tests/scripts/patch-corefx.py

```diff
@@ -0,0 +1,307 @@
+                       'jitinterface.dll',
+                       'mscorrc.debug.dll',
+                       'mscorrc.dll',
+                       'sos.dll']
+
+    test_log('forbidden_names = %s' % forbidden_names)
+
+    for file in os.listdir(core_root):
+        test_log('considering file %s' % file)
+
+        filename = os.path.basename(file)
+        comparename = filename.lower()
+        
+        if (    comparename.endswith('.dll') and  
+                comparename not in forbidden_names and
+                not comparename.startswith('api-ms-core')   ):
+            dll_names.append(filename)
+            test_log('adding file %s to list of patchable files' % filename)
+
+    return dll_names
+
+def patch_coreclr_root(core_root, fx_bin, corefx_binaries):
+    """ Walk through the fx bin and patch any dlls listed in
+        corefx_binaries to the core root.
+    Args:
+        core_root       (str):       the core root path
+        fx_bin          (str):       the runtime folder from a corefx build
+        corefx_binaries (list(str)): the list of binaries to patch
+    Returns:
+        nothing
+    """
+    test_log('Patching coreclr core_root')
+
+    for file in os.listdir(fx_bin):
+        test_log('considering file %s' % file)
+
+        filename = os.path.basename(file)
+        if filename in corefx_binaries:
+            source_pathname = os.path.join(fx_bin, filename)
+            target_pathname = os.path.join(core_root, filename)
+
+            test_log ('copying file %s to file %s' % (source_pathname, target_pathname))
+
+            shutil.copy2(source_pathname, target_pathname)
+
+##########################################################################
+# Main
+##########################################################################
+
+def main(args):
+    """
+        The way this script decides what to patch is by looking at the core
+        root for a list of dlls, then filtering out any ones built by coreclr.
+        This leaves us with a list of non-coreclr build dlls. Now we can use
+        that list to go through the corefx repo and any ones that also exist
+        in the corefx bin folder are copied over.
+    """
+
+    log('Patching CoreFX binaries from local enlistment.')
+
+    arch, build_type, clr_core_root, fx_root = validate_args(args)
+
+    clr_os = 'Windows_NT' if Is_windows else Unix_name_map[os.uname()[0]]
+
+    if not os.path.exists(clr_core_root):
+        raise Exception('Core root path %s does not exist.' % (core_root))
+
+    corefx_binaries = walk_core_root(clr_core_root)
+
+    fx_bin = os.path.join(fx_root,
+                          'artifacts',
+                          'bin',
+                          'runtime',
+                          'netcoreapp-%s-%s-%s' % (clr_os, build_type, arch))
+
+    if not os.path.exists(fx_bin):
+        raise Exception('CoreFX bin path %s does not exist.' % (core_root))
+
+    patch_coreclr_root(clr_core_root, fx_bin, corefx_binaries)
```

Comment:
This seems to be creating the core_root under the assumption that there were previously restored files which we will overwrite. This is an incorrect assumption, we need to be able to generate the core_root cleanly without package restore.

Reply:
I can assume that core_root will exist though, right? Since even without restore we will copy coreclr and its dependencies in to core_root.

My script does depend on restore, because it only copies files from corefx that already exist in core_root. Removing that dependency should be fairly trivial though.